opengrok-indexer/src/main/java/org/opengrok/indexer/index/IndexDatabase.java

```diff
@@ -1963,12 +1973,11 @@ private void indexParallel(String dir, IndexDownArgs args) {
 
         // Start with failureCount=worksCount, and then subtract successes.
         int failureCount = worksCount;
-        if (bySuccess != null) {
-            List<IndexFileWork> successes = bySuccess.getOrDefault(Boolean.TRUE, null);
-            if (successes != null) {
-                failureCount -= successes.size();
-            }
+        List<IndexFileWork> successes = bySuccess.getOrDefault(Boolean.TRUE, null);
+        if (successes != null) {
```

Comment:
We can replace this if condition with
failureCount -= Optional.ofNullable(bySuccess.get(Boolean.TRUE))
                    .map(List::size)
                    .orElse(0);

Reply:
Nice. The `successes` definition can go away too. Thanks.